vision_project/test_notebook.ipynb

```diff
@@ -9,7 +9,7 @@
   },
```

Comment:
i am getting an error with densenet

ValueError                                Traceback (most recent call last)
Cell In[9], [line 8](vscode-notebook-cell:?execution_count=9&line=8)
      [5](vscode-notebook-cell:?execution_count=9&line=5) input_shape = (224,224,3)
      [7](vscode-notebook-cell:?execution_count=9&line=7) # Create an instance of the model
----> [8](vscode-notebook-cell:?execution_count=9&line=8) model = DenseNetVisionModel(num_classes=25, input_shape=input_shape, weights='imagenet')
     [10](vscode-notebook-cell:?execution_count=9&line=10) # Compile the model
     [11](vscode-notebook-cell:?execution_count=9&line=11) model.compile(optimizer='adam', loss='categorical_crossentropy', metrics=['accuracy'])

File ~/advvision-lumbar-spine-classification/vision_project/vision_models/densenetmodel.py:77, in DenseNetVisionModel.__init__(self, num_classes, input_shape, weights)
     [75](https://vscode-remote+ssh-002dremote-002bvision-002dproject-002dvm-002eus-002dcentral1-002df-002epk-002darg-002dprj4-002ddatasci.vscode-resource.vscode-cdn.net/home/kwaku/advvision-lumbar-spine-classification/vision_project/~/advvision-lumbar-spine-classification/vision_project/vision_models/densenetmodel.py:75)     self.input_shape = input_shape
     [76](https://vscode-remote+ssh-002dremote-002bvision-002dproject-002dvm-002eus-002dcentral1-002df-002epk-002darg-002dprj4-002ddatasci.vscode-resource.vscode-cdn.net/home/kwaku/advvision-lumbar-spine-classification/vision_project/~/advvision-lumbar-spine-classification/vision_project/vision_models/densenetmodel.py:76) else:
---> [77](https://vscode-remote+ssh-002dremote-002bvision-002dproject-002dvm-002eus-002dcentral1-002df-002epk-002darg-002dprj4-002ddatasci.vscode-resource.vscode-cdn.net/home/kwaku/advvision-lumbar-spine-classification/vision_project/~/advvision-lumbar-spine-classification/vision_project/vision_models/densenetmodel.py:77)     raise ValueError(f"Unexpected input shape: {input_shape}")
     [79](https://vscode-remote+ssh-002dremote-002bvision-002dproject-002dvm-002eus-002dcentral1-002df-002epk-002darg-002dprj4-002ddatasci.vscode-resource.vscode-cdn.net/home/kwaku/advvision-lumbar-spine-classification/vision_project/~/advvision-lumbar-spine-classification/vision_project/vision_models/densenetmodel.py:79) self.slices = self.input_shape[0]
     [80](https://vscode-remote+ssh-002dremote-002bvision-002dproject-002dvm-002eus-002dcentral1-002df-002epk-002darg-002dprj4-002ddatasci.vscode-resource.vscode-cdn.net/home/kwaku/advvision-lumbar-spine-classification/vision_project/~/advvision-lumbar-spine-classification/vision_project/vision_models/densenetmodel.py:80) self.image_shape = self.input_shape[1:]

ValueError: Unexpected input shape: (224, 224, 3)

Reply:
I played around with the code and everything all you need to do is add this to the last cell and its good to go

```
from vision_models.densenetmodel import ModelTrainer
from trainer import VisionModelPipeline

# Use the original model
trainer = ModelTrainer(model)
pipeline = VisionModelPipeline()
pipeline.setup_environment()

study_ids = ['4003253','8785691', '7143189','4646740']
train_dataset = pipeline.load_data("train", study_ids)
val_dataset = pipeline.load_data("val")

history = trainer.train(train_dataset, val_dataset, epochs=5)

for img, labels in train_dataset.take(1):
    print(img.shape)
    print(labels.shape)
    print(labels)
```

Reply:
PR#11 will take care of this.